### DISCUSSION TITLE
Statistics generation and analysis (Hibernate, Infinispan)

### DISCUSSION THREAD

**Discussion by dm-vestmark** [infinispan/infinispan#11649] [General]
We want guidance on two issues pertaining to hibernate statistics in a clustered environment.  
     - How to generate statistics for a node in the cluster and not cluster-wide
     - How to use these statistics to understand trends in cache usage 

We are looking into an apparent sporadic cache performance degradation over a period of several days in a production environment. We are able to generate statistics at the cluster level using the following additions to our config files: 

`standalone-ha.xml` : 
            `<cache-container` name="VmManagedBatchedMda" marshaller="JBOSS" modules="org.infinispan.hibernate-cache" `statistics-enabled="true">`
                `<invalidation-cache` name="entity" remote-timeout="120000" `statistics-enabled="true">`
                   `` <heap-memory `size="100"/>`
                   `` <expiration max-idle="300000" `lifespan="300000"/>`
                `</invalidation-cache>`

and setting hibernate.cache.infinispan.statistics to true.
  "[hibernate.cache.infinispan.statistics]": true

1. How to generate statistics for a node in the cluster and not cluster-wide
These additions have produced only statistics at the cluster level and not the local node level.  The cluster-wide statistics do not allow us to analyze why a specific node(s) experience performance degradation based on cache usage.  Thus, we need to generate and analyze statistics at the node level.
When we do this in a non-clustered environment (with standalone.xml and a <local-cache>), we get a properly working Statistics mbean for each Hibernate entity. We get a properly working ClusterCacheStats bean in a cluster, but the values are the same on each cluster node. The Statistics mbean is still there, but we get an org.infinispan.commons.CacheException: java.lang.NullPointerException trying to access every single JMX attribute from the Statistics bean. Are the local Statistics supposed to work for an invalidation cache and report the local statistics (hits, misses, evictions) for each cluster node? And how does the ClusterCacheStats aggregate the numbers – is it adding up the hits, misses, evictions, and invalidations across all cluster nodes and reporting the sum?
2. How to use these statistics to understand trends in cache usage 
The statistics also need to be captured for usage analysis over time. Is there some easy way to log the statistics over time for multiple entities? Or is the only way to configure our JMX tools to capture and track multiple attributes for multiple mbeans corresponding to the various entities we are interested in?
The Environment: Wildfly version: 26.0.1.Final
Inifinispan version: 12.1.7.Final
Hibernate version: 5.4.18.Final

**Comment by dm-vestmark** [comment]
Any assistance would be much appreciated

**Comment by tristantarrant** [comment]
Closing this as a duplicate of https://github.com/infinispan/infinispan/discussions/11650
